source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRFactSetRenderer.mm

```diff
@@ -39,22 +39,19 @@ + (ACRUILabel *)buildLabel:(NSString *)text
     lab.isFactSetLabel = YES;
     lab.translatesAutoresizingMaskIntoConstraints = NO;
     lab.style = style;
-    __block NSMutableAttributedString *content = nil;
+    NSMutableAttributedString *content = nil;
     if(rootView){
+        std::shared_ptr<FactSet> fctSet = std::dynamic_pointer_cast<FactSet>(element);
         NSMutableDictionary *textMap = [rootView getTextMap];
-        // Syncronize access to imageViewMap
-        dispatch_sync([rootView getSerialTextQueue], ^{
-            if(textMap[elementId]) { // if content is available, get it, otherwise cache label, so it can be used used later
-                content = textMap[elementId];
-            } else {
-                textMap[elementId] = lab;
-            }
-        });
-    }
-
-    if(content){
+        NSDictionary* data = textMap[elementId];
```

Comment:
>textMap[elementId] [](start = 29, length = 18)

I'm a little confused about the synchronization. Elsewhere it looks like access to this textMap is serialized on ACRView's _serial_text_queue, but here we're accessing it from a different thread without any synchronization.

Reply:
yes, the reason is that once asyncronous tasks are completed, remaining works are done in serial since the works are UI related and have to run on main thread.

---
In reply to: [193501004](https://github.com/Microsoft/AdaptiveCards/pull/1547#discussion_r193501004) [](ancestors = 193501004)